.docker/Dockerfile-CPU-Base

```diff
@@ -3,38 +3,144 @@
 # ------------------------------------------------------------------
 # Ubuntu           18.04
 # OpenMPI          latest       (apt)
-# cmake            3.10         (git)
-# MKL              2018.4-057   (apt)
-# arrayfire        3.7.1        (git, CPU backend)
-# oneDNN           2.0          (git)
-# Gloo             b7e0906      (git)
-# libsndfile       4bdd741      (git)
+# cmake            3.10         (oficial binaries)
+# MKL              2020.4-912   (apt)
+# arrayfire        3.7.3        (git, CPU backend)
+# libsndfile       latest       (apt, v1.0.28-4)
+# oneDNN           master       (git)
+# Gloo             1da2117      (git)
 # FFTW             latest       (apt)
 # KenLM            4a27753      (git)
 # GLOG             latest       (apt)
 # gflags           latest       (apt)
-# python           3.6          (apt)
+# python3          latest       (apt)
+# boost            1.75.0       (source)
 # ==================================================================
 
-FROM ubuntu:18.04
+FROM ubuntu:18.04 as cpu_base_builder
 
-ENV APT_INSTALL="apt-get install -y --no-install-recommends"
-ENV MKLROOT="/opt/intel/mkl"
+ENV DEBIAN_FRONTEND=noninteractive
 
-RUN apt-get update && \
-    DEBIAN_FRONTEND=noninteractive $APT_INSTALL \
+RUN apt-get update && apt-get install -y --no-install-recommends \
         build-essential \
         ca-certificates \
-        wget \
+        curl \
+        git \
+        g++
+        # for cmake
+        #zlib1g-dev libcurl4-openssl-dev \
+        # for MKL
+        #apt-transport-https \
```

Comment:
are we not installing MKL? I also see CBLAS install cmds below

Reply:
we are not installing for image which used in some deps installation, but directly install in the final image. More clean version find in https://github.com/facebookresearch/flashlight/pull/401.